### PR TITLE
Make paasta status for adhoc jobs use HTTP API

### PR DESCRIPTION
**Pull request by EmanekaT**
Tested manually https://fluffy.yelpcorp.com/i/cGB6CXkFd8P1WFV5FjRRcL8zjtRWqWHt.html
I didn't add any tests as there were any tests at the first place. It's mostly just parsing. 
I also fixed a comment for make dev-api.

### CODE REVIEW

File: paasta_tools/api/api_docs/swagger.json [hunk 1]
```diff
@@ -880,6 +884,31 @@
...
+      "items": {
+        "$ref": "#/definitions/AdhocLaunchHistory"
+      }
+    },
+    "AdhocLaunchHistory": {
+        "type": "object",
+        "description": "A single run",
+        "properties": {
+           "launch time": {
+              "type": "string",
```
Comment (solarkennedy) [inline paasta_tools/api/api_docs/swagger.json]: Just a small preference here, but can you use underscores `launch_time`, etc to be consistent with the variables names any everything?

**Comment by solarkennedy** [comment]
Keep in mind that this needs to be a two phase deploy to be "correct".

The first phase needs to be the API changes, *Then* the client changes.

Otherwise users will get errors because their client will be contacting a server that has no idea about these things.

Or just deploy in the evening, which would be fine. The important part is to not have this "half" deployed. (some clusters we pin)

**Comment by stug** [comment]
I guess since this is already all being done in a single review, let's just release this in the evening sometime next week -- we'll need to make a paasta release and then a puppet change to update the pinned version in the WYNR clusters.

**Comment by EmanekaT** [comment]
> I guess since this is already all being done in a single review, let's just release this in the evening sometime next week -- we'll need to make a paasta release and then a puppet change to update the pinned version in the WYNR clusters.

Sounds good

File: paasta_tools/cli/cmds/status.py [hunk 2]
```diff
@@ -73,10 +73,10 @@
 HTTP_ONLY_INSTANCE_CONFIG: Sequence[Type[InstanceConfig]] = [
     FlinkClusterConfig,
     KubernetesDeploymentConfig,
+    AdhocJobConfig,
```
[the remark this answers is not in the document]
Comment (stug) [reply paasta_tools/cli/cmds/status.py]: (also I added another ticket to the epic to clean up paasta_serviceinit when we're sure of our implementations)

File: paasta_tools/cli/cmds/status.py [hunk 3]
```diff
@@ -574,8 +601,8 @@ def report_status_for_cluster(
             deployed_instance
             for deployed_instance in deployed_instances
             if (
-                deployed_instance in ssh_only_instances
-                or deployed_instance not in http_only_instances and not use_api_endpoint
+                deployed_instance in ssh_only_instances or
```
[the remark this answers is not in the document]
Comment (EmanekaT) [reply paasta_tools/cli/cmds/status.py]: Yes.

File: paasta_tools/cli/cmds/status.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (EmanekaT) [reply paasta_tools/cli/cmds/status.py]: I'm confident with it is now. This code is pretty simple and I have tested it already.

**Pull request merged**